File: Programmers/Inho/Level01/2016년.cs
using System;
public class Solution
{
    public string solution(int a, int b)
    {
        string answer = "";
        int days = (a - 1) * 31 + b;
        switch (a)
        {
            case 12:
                days -= 6;
                break;
            case 11:
                days -= 5;
                break;
            case 10:
                days -= 5;
                break;
            case 9:
                days -= 4;
                break;
            case 8:
                days -= 4;
                break;
            case 7:
                days -= 4;
                break;
            case 6:
                days -= 3;
                break;
            case 5:
                days -= 3;
                break;
            case 4:
                days -= 2;
                break;
            case 3:
                days -= 2;
                break;
        }
        days = days % 7;
        switch (days)
        {
            case 1:
                return answer = "FRI";
            case 2:
                return answer = "SAT";
            case 3:
                return answer = "SUN";
            case 4:
                return answer = "MON";
            case 5:
                return answer = "TUE";
            case 6:
                return answer = "WED";
            case 0:
                return answer = "THU";
        }

        return answer;
    }
}

File: Programmers/Inho/C++/Level01/2016년.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(int a, int b) {
    int mon[12] = {31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    string days[7] = {"FRI", "SAT", "SUN","MON","TUE","WED","THU"};
    string answer = "";
    int day = 0;
    for(int i = 0;i<a-1;i++){
        day += mon[i];
    }
    day+=b;
    
    answer = days[(day-1)%7] ;
    return answer;
}